controllers/tutorial.controller.js

```diff
@@ -0,0 +1,73 @@
+const Tutorial = require("../models/tutorial.schema");
+
+// get All Tutorials
+const tutorialsAll = async (req, res) => {
+  try {
+    const tutorials = await Tutorial.find().sort({
+      createdAt: -1,
+    });
+    res.json(tutorials);
+  } catch (error) {
+    res.json({ message: error });
+  }
+};
+
+// Add new Tutorials
+const tutorialsCreate = async (req, res) => {
+  const tutorial = new Tutorial({
+    published: req.body.published,
+  });
+  try {
+    const savedTutorial = await tutorial.save();
+    res.send(savedTutorial);
+  } catch (error) {
+    res.status(400).send(error);
+  }
+};
+
+// Update a Tutorial by ID
+const updateTutorials = async (req, res) => {
+  try {
+    const { update } = req.body;
+    const { options } = { new: true };
+    const tutorialUpdate = await Tutorial.findByIdAndUpdate(
+      { _id: req.params.tutorialId },
+      { id: req.body.id },
+      update,
+      options,
+    );
+    res.json(tutorialUpdate);
+  } catch (error) {
+    res.status(400).send(error);
```

Comment:
- Log the error
- Should be 500 response with `{ message: 'Internal Server Error' }`.